test/test_jit.py

```diff
@@ -1876,6 +1876,57 @@ def test_print_graph_executor(meme):
         if not WINDOWS:
             self.assertExpected(captured[0])
 
+    def test_script_module(self):
+        class M1(torch.jit.ScriptModule):
+            def __init__(self):
+                super(M1, self).__init__(False)
+                self.weight = nn.Parameter(torch.randn(2))
+
+            @torch.jit.script_method
+            def forward(self, thing):
+                return self.weight + thing
+
+        class M2(torch.jit.ScriptModule):
+            def __init__(self):
+                super(M2, self).__init__(False)
+                # test submodule
+                self.sub = M1()
+                # test parameters
+                self.weight = nn.Parameter(torch.randn(2, 3))
+                self.bias = nn.Parameter(torch.randn(2))
+                # test defining a method from a string
+                self.define("""
+                    def hi(self, a):
+                        return self.weight.mm(a)
+                """)
+            # test script methods
+
+            @torch.jit.script_method
```

Comment:
Can we have a test for a non-scripted method in a scripted module?

Reply:
That currently is not supported :), Ill implement it in the next PR that makes script module a nn.Module.